Sources/Apollo/ApolloInterceptor.swift

```diff
@@ -4,6 +4,8 @@ import ApolloAPI
 
 /// A protocol to set up a chainable unit of networking work.
 public protocol ApolloInterceptor {
+
+  var id: String { get set }
```

Comment:
`Identifiable` is only available on iOS 13+ and macOS 15+, we support lower versions of both. This imposes the same 'identifiable' requirement for interceptors.

Reply:
1. It would be really helpful to document this method and explain what the purpose and scope of this identifier is, so that custom implementers can decide what the best identifier would be for their specific use case.
2. In order to avoid the breaking change introduced by this, have you considered a default implementation of this requirement using a protocol extension?

Reply:
> It would be really helpful to document this method and explain what the purpose and scope of this identifier is, so that custom implementers can decide what the best identifier would be for their specific use case.

Documentation can be added to clarify it's purpose, I'll get that done. As long as it's unique to other identifiers and a `String` it shouldn't matter. This was required as a workaround to the interceptor architecture while fixing the memory leak and maintaining the ability to support HTTP-based subscriptions.

> In order to avoid the breaking change introduced by this, have you considered a default implementation of this requirement using a protocol extension?

Yes I did, but Swift extensions cannot hold stored properties and the identifier needs to be constant during the lifetime of the interceptor. Static constants don't work here either.

Reply:
> Documentation can be added to clarify it's purpose, I'll get that done.

Thanks! Perhaps the scope is the interceptor instance? What does "other identifiers" mean concretely? Unique within the set of installed interceptors active during the processing of one request?

> Swift extensions cannot hold stored properties and the identifier needs to be constant during the lifetime of the interceptor.

Ahh, makes sense. Perhaps an alternative might've been an `id { ObjectIdentifier(self) }`, but it would incur a protocol requirement for class-semantics by the implementer.

Reply:
> What does "other identifiers" mean concretely? Unique within the set of installed interceptors active during the processing of one request?

Correct. The interceptor provider passed into `RequestChainNetworkTransport` during initialization is used to build the [request chain for each request](https://github.com/apollographql/apollo-ios/blob/main/Sources/Apollo/RequestChainNetworkTransport.swift#L130). They need to be unique amongst that array of interceptors.